cockatrice/src/client/ui/widgets/deck_editor/deck_editor_filter_dock_widget.cpp

```diff
@@ -0,0 +1,142 @@
+#include "deck_editor_filter_dock_widget.h"
+
+#include "../../../../game/cards/card_database_model.h"
+#include "../../../../game/filters/filter_builder.h"
+#include "../../../../game/filters/filter_tree_model.h"
+#include "../../../../settings/cache_settings.h"
+
+#include <QGridLayout>
+#include <QMenu>
+#include <QToolButton>
+
+DeckEditorFilterDockWidget::DeckEditorFilterDockWidget(QWidget *parent, TabGenericDeckEditor *_deckEditor)
+    : QDockWidget(parent), deckEditor(_deckEditor)
+{
+    setObjectName("filterDock");
+
+    setFeatures(QDockWidget::DockWidgetClosable | QDockWidget::DockWidgetFloatable | QDockWidget::DockWidgetMovable);
+
+    createFiltersDock();
+}
+
+void DeckEditorFilterDockWidget::createFiltersDock()
+{
+    filterModel = new FilterTreeModel();
+    filterModel->setObjectName("filterModel");
+    deckEditor->databaseDisplayDockWidget->databaseDisplayModel->setFilterTree(filterModel->filterTree());
```

Comment:
Would these be the kind of things we'd want to hide behind methods in the top level obj or secondary level obj compared to chaining pointers in such a way? I just worry this could be unsafe since we don't check these pointers